### DISCUSSION TITLE
Google Gemini support?

### DISCUSSION THREAD

**Discussion by jaygz316** [karakeep-app/karakeep#453] [Ideas]
Is there any chance there might be support for Gemini in the future? I have come to really like it and I already have my stuff in one dashboard, just wondering if there was a chance.

**Comment by kamtschatka** [comment]
probably makes sense to switch to something like https://github.com/themaximalist/llm.js or https://github.com/samestrin/llm-interface, since everyone is thinking that their LLM is better than the other anyway and we'll not add support for every single one of them.

**Comment by hobybrenner** [comment]
Also, supporting this might be an additional solution. The AI Gateway streamlines requests to 250+ language, vision, audio and image models with a unified API.
[https://github.com/Portkey-AI/gateway/](url)

**Comment by Inrego** [comment]
Another reason to support Gemini is that it has a pretty generous free tier (free as in you pay with data instead of money)

**Comment by MohamedBassem** [comment]
did anyone try this: https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/call-vertex-using-openai-library

Seems like gemini has openai compatible API. You just need to set the correct base url and it'll work.

**Comment by lucius346346** [comment]
I have problem with setting this up? Should it be:
OPENAI_BASE_URL=https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash-latest:generateContent
OPENAI_API_KEY={api key}

**Reply by sebp** [reply]
I got it working using

```
OPENAI_BASE_URL="https://generativelanguage.googleapis.com/v1beta/"
OPENAI_API_KEY="XXXXXXXX"
INFERENCE_TEXT_MODEL="gemini-1.5-flash-8b"
INFERENCE_IMAGE_MODEL="gemini-1.5-flash-8b"
```

In addition, I had to add the following tagging rule under User Settings → AI Settings

```
You must respond in JSON using this JSON schema:
     Tags = {"tags": list[str]}
     Return: Tags
```

**Reply by lucius346346** [reply]
> I got it working using
> 
> ```
> OPENAI_BASE_URL="https://generativelanguage.googleapis.com/v1beta/"
> OPENAI_API_KEY="XXXXXXXX"
> INFERENCE_TEXT_MODEL="gemini-1.5-flash-8b"
> INFERENCE_IMAGE_MODEL="gemini-1.5-flash-8b"
> ```
> 
> In addition, I had to add the following tagging rule under User Settings → AI Settings
> 
> ```
> You must respond in JSON using this JSON schema:
>      Tags = {"tags": list[str]}
>      Return: Tags
> ```

Awesome! Thanks a lot!

**Reply by Trinkes** [reply]
Does this still work?

**Reply by lucius346346** [reply]
> Does this still work?

It does for me.

**Reply by Trinkes** [reply]
It's now working for me as well. For some reason my .env file is not being loaded correctly but when I add the variables directly in docker compose file it works. Must have some kind of miss configuration. 

It's working now, thank you!

**Reply by ianhouser** [reply]
> I got it working using
> 
> ```
> OPENAI_BASE_URL="https://generativelanguage.googleapis.com/v1beta/"
> OPENAI_API_KEY="XXXXXXXX"
> INFERENCE_TEXT_MODEL="gemini-1.5-flash-8b"
> INFERENCE_IMAGE_MODEL="gemini-1.5-flash-8b"
> ```
> 
> In addition, I had to add the following tagging rule under User Settings → AI Settings
> 
> ```
> You must respond in JSON using this JSON schema:
>      Tags = {"tags": list[str]}
>      Return: Tags
> ```

I believe you can avoid adding the tagging rule by setting this [environment variable](https://docs.karakeep.app/configuration) to "json": `INFERENCE_OUTPUT_SCHEMA`. The default is "structured". It seems to work for me at least.

Also noting that adding the additional environment variables above in the .env was not working - karakeep kept overriding and using OpenAI. But inserting them into the docker-compose.yml environment worked perfectly.

Lastly, just an obvious FYI that you can update inference variables to match newer models. I'm currently using: `gemini-2.5-flash-preview-05-20`
You can find the latest models and their names [here](https://cloud.google.com/vertex-ai/generative-ai/docs/models/gemini/2-5-flash).

**Comment by jbohnslav** [comment]
If you self host a litellm proxy, you should be able to use whatever LLM you want.

**Reply by hannibalshosting88** [reply]
anyone confirmed this functionality?

**Comment by liveisjoy** [comment]
Anyone  had managed to implement this feature so that Gemini can be used in Hoarder?

**Reply by Trinkes** [reply]
I use gemini without any issue

**Comment by DieserTii** [comment]
I use it with Gemini and it works really good!
